Could not find artifact `io.micronaut:micronaut-runtime:jar:netty` with Maven application

### Expected Behavior

It should be possible to run the application with Maven.

### Actual Behaviour

Using Maven and Micronaut `3.1.2-SNAPSHOT` fails to resolve one artifact:

```
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-runtime/netty/micronaut-runtime-netty.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.139 s
[INFO] Finished at: 2021-10-25T09:55:40+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project maven: Could not resolve dependencies for project example.micronaut:maven:jar:0.1: Could not find artifact io.micronaut:micronaut-runtime:jar:netty in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

I think the error is related to this PR #6404 

### Steps To Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-maven-graal`
- `cd micronaut-maven-graal`
- `git checkout 3.1.x`
- `./mvnw mn:run`

### Environment Information

Linux Mint 20.2 & Jdk 8

### Example Application

_No response_

### Version

3.1.2-SNAPSHOT

--- a/bom/build.gradle
+++ b/bom/build.gradle
@@ -116,7 +116,7 @@ publishing {
                                     && it.groupId.text() == group
                         }
                         if (pomDep != null) {
-                            pomDep.version.first().setValue("\${${propertyName}}")
+                            pomDep.version.first().setValue("\${${propertyName}.version}")
                         } else {
                             println "[WARNING] Didn't find dependency ${group}:micronaut-${p.name} in BOM file"
                         }
@@ -142,7 +142,7 @@ publishing {
                 if (excludedProjects.contains(p.name)) {
                     continue
                 }
-                String propertyName = "micronaut.${p.name.replace((char) '-', (char) '.')}"
+                String propertyName = "micronaut.${p.name.replace((char) '-', (char) '.')}.version"
                 pom.properties.put(propertyName, p.version)
             }
 

